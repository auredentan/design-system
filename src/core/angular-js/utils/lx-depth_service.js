function LxDepthService() {
    const service = this;

    /////////////////////////////
    //                         //
    //    Private attributes   //
    //                         //
    /////////////////////////////

    /**
     * The initial depth.
     *
     * @type {number}
     */
    // eslint-disable-next-line no-magic-numbers
    let _depth = 1000;

    /////////////////////////////
    //                         //
    //     Public functions    //
    //                         //
    /////////////////////////////

    /**
     * Get the highest depth.
     *
     * @return {number} The highest depth.
     */
    function get() {
        return _depth;
    }

    /**
     * Increase depth.
     */
    function increase() {
        _depth++;
    }

    /////////////////////////////

    service.get = get;
    service.increase = increase;
}

/////////////////////////////

angular.module('lumx.utils.depth').service('LxDepthService', LxDepthService);

/////////////////////////////

export { LxDepthService };
