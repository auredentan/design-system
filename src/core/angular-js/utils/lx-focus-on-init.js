function lxFocusOnInitDirective() {
    function link(scope, el) {
        el.focus();
    }

    return {
        link,
    };
}

/////////////////////////////

angular.module('lumx.utils.focus-on-init').directive('lxFocusOnInit', lxFocusOnInitDirective);

/////////////////////////////

export { lxFocusOnInitDirective };
