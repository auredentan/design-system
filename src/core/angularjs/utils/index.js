/* eslint-disable import/no-unassigned-import */

import './depth_service';
import './enter-keypress_directive';
import './event-scheduler_service';
import './focus-on-init_directive';
import './focus-trap_service';
import './utils_service';
