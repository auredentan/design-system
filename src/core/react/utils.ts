import React from 'react';

import get from 'lodash/get';
import has from 'lodash/has';
import isEmpty from 'lodash/isEmpty';
import isString from 'lodash/isString';

/////////////////////////////

type Component = string | React.FC<any> | React.PureComponent<any, any> | React.Component<any, any>;

/////////////////////////////

/**
 * Check if a ReactElement is of the given type.
 *
 * @param  {React.ReactElement} el   The ReactElement we want to check the type of.
 * @param  {string|Component}   type The type we want to check if the ReactElement is of.
 * @return {boolean}            If the ReactElement is of the given type or not.
 */
function isElementOfType(el: React.ReactElement, type: Component): boolean {
    if (!isString(type) && has(type, 'displayName')) {
        type = get(type, 'displayName', get(type, '_reactInternalFiber.elementType.name', 'Component'));
    }

    if (isEmpty(type) && !isString(type)) {
        throw new Error(
            `The type you want to check is not valid. Waiting a component or a string, got ${type.toString()}`,
        );
    }

    if (type === 'text') {
        return isString(el) || el.type === 'span';
    }

    return get(el.type, 'name') === type;
}

/**
 * Check if a ReactElement is a text (i.e. either a pure text node or a <span>).
 * Simply check if the ReactElement is a string or if its type is 'span'.
 *
 * @param  {React.ReactElement} el The ReactElement to check if it's a text.
 * @return {boolean}            If the ReactElement is a text or not.
 */
function isElementText(el: React.ReactElement): boolean {
    return isElementOfType(el, 'text');
}

/////////////////////////////

export { Component, isElementOfType, isElementText };
