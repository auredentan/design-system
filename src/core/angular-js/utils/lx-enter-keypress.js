import { MODULE_NAME } from '@lumx/angularjs/lumx'

/////////////////////////////

function lxEnterKeypressDirective() {
    function link(scope, el, attrs) {
        const _ENTER_KEY_CODE = 13;

        el.bind('keydown keypress', function onKeyPress(evt) {
            if (evt.which === _ENTER_KEY_CODE) {
                scope.$apply(function evalExpression() {
                    scope.$eval(attrs.lxEnterKeypress, { $event: evt });
                });

                evt.preventDefault();
            }
        });
    }

    return {
        link,
    };
}

/////////////////////////////

angular.module(`${MODULE_NAME}.utils.enter-keypress`).directive('lxEnterKeypress', lxEnterKeypressDirective);

/////////////////////////////

export { lxEnterKeypressDirective };
