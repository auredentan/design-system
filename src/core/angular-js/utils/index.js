import './lx-depth_service';
import './lx-enter-keypress';
import './lx-event-scheduler_service';
import './lx-utils_service';
