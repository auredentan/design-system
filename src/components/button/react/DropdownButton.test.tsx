import React, { Fragment } from 'react';

import { mount, shallow } from 'enzyme';
import mockConsole from 'jest-mock-console';
import { build, fake, oneOf } from 'test-data-bot';

import { Button, ButtonGroup, ButtonVariants, Icon } from 'LumX';
import { ICommonSetup, Wrapper, commonTestsSuite } from 'LumX/core/testing/utils.test';
import { mdiPlus } from 'LumX/icons';

import { CLASSNAME, DropdownButton, DropdownButtonProps, Emphasises, Sizes, Themes } from './DropdownButton';

/////////////////////////////

/**
 * Define the overriding properties waited by the `setup` function.
 */
type ISetupProps = Partial<DropdownButtonProps>;

/**
 * Defines what is returned by the setup function.
 */
interface ISetup extends ICommonSetup {
    /**
     * The properties of the tested component.
     */
    props: ISetupProps;

    /**
     * The <Button> element that is used as a wrapper for the children of the <DropdownButton>.
     */
    button: Wrapper;

    /**
     * The <Dropdown> element that holds the dropdown toggled by the <DropdownButton>.
     */
    dropdown: Wrapper;

    /**
     * The <ButtonGroup> element that is used as a wrapper for the buttons inside of the <DropdownButton> when
     * splitted.
     */
    group: Wrapper;

    /**
     * The <Icon>(s) that holds either the preceding icon (the one before the label) or the dropdown toggle icon.
     */
    icon: Wrapper;

    /**
     * The <IconButton> that hold the dropdown toggle icon button.
     */
    iconButton: Wrapper;

    /**
     * The roo element of the <DropdownButton> component.
     * Either <ButtonGroup> if `isSplitted` or <Button> if not `isSplitted`.
     */
    root: Wrapper;
}

/////////////////////////////

/**
 * Mounts the component and returns common DOM elements / data needed in multiple tests further down.
 *
 * @param  {ISetupProps} props                   The props to use to override the default props of the component.
 * @param  {boolean}     [shallowRendering=true] Indicates if we want to do a shallow or a full rendering.
 * @return {ISetup}      An object with the props, the component wrapper and some shortcut to some element inside of
 *                       the component.
 */
const setup: (props?: ISetupProps, shallowRendering?: boolean) => ISetup = (
    { ...propsOverrides }: ISetupProps = {},
    shallowRendering: boolean = true,
): ISetup => {
    const props: DropdownButtonProps = {
        children: 'Label',
        dropdown: <span>Dropdown content</span>,
        ...propsOverrides,
    };

    const renderer: (el: JSX.Element) => Wrapper = shallowRendering ? shallow : mount;

    const wrapper: Wrapper = renderer(<DropdownButton {...props} />);

    return {
        root: props.isSplitted ? wrapper.find('ButtonGroup') : wrapper.find('Button'),

        group: wrapper.find('ButtonGroup'),

        button: wrapper.find('Button'),
        icon: wrapper.find('Icon'),
        iconButton: wrapper.find('IconButton'),

        dropdown: wrapper.find('Dropdown'),

        props,
        wrapper,
    };
};

describe(`<${DropdownButton.displayName}>`, () => {
    // 1. Test render via snapshot (default state of component).
    describe('Snapshots and structure', (): void => {
        it('should render correctly a non-splitted dropdown button', (): void => {
            const { button, dropdown, group, icon, iconButton, root, wrapper }: ISetup = setup();
            expect(wrapper).toMatchSnapshot();

            // tslint:disable-next-line: no-non-null-assertion
            expect(root).toHaveDisplayName(Button.displayName!);

            expect(group).not.toExist();

            expect(button).toExist();
            expect(button).toHaveClassName(CLASSNAME);

            expect(iconButton).not.toExist();

            expect(icon).toExist();
            expect(icon.length).toEqual(1);

            expect(dropdown).not.toExist();
        });

        it('should render correctly a non-splitted dropdown button with a preceding icon', (): void => {
            const children: React.ReactNode = (
                <Fragment>
                    <Icon icon={mdiPlus} />
                    Label
                </Fragment>
            );

            const { button, dropdown, group, icon, iconButton, root, wrapper }: ISetup = setup({ children });
            expect(wrapper).toMatchSnapshot();

            // tslint:disable-next-line: no-non-null-assertion
            expect(root).toHaveDisplayName(Button.displayName!);

            expect(group).not.toExist();

            expect(button).toExist();
            expect(button).toHaveClassName(CLASSNAME);

            expect(iconButton).not.toExist();

            expect(icon).toExist();
            expect(icon.length).toEqual(2);

            expect(dropdown).not.toExist();
        });

        it('should render correctly a splitted dropdown button', (): void => {
            const { button, dropdown, group, icon, iconButton, root, wrapper }: ISetup = setup({ isSplitted: true });
            expect(wrapper).toMatchSnapshot();

            // tslint:disable-next-line: no-non-null-assertion
            expect(root).toHaveDisplayName(ButtonGroup.displayName!);

            expect(group).toExist();
            expect(group).toHaveClassName(CLASSNAME);

            expect(button).toExist();
            expect(iconButton).toExist();
            expect(icon).toExist();
            expect(icon.length).toEqual(1);

            expect(dropdown).not.toExist();
        });

        it('should render correctly a splitted dropdown button with a preceding icon', (): void => {
            const children: React.ReactNode = (
                <Fragment>
                    <Icon icon={mdiPlus} />
                    Label
                </Fragment>
            );

            const { button, dropdown, group, icon, iconButton, root, wrapper }: ISetup = setup({
                children,
                isSplitted: true,
            });
            expect(wrapper).toMatchSnapshot();

            // tslint:disable-next-line: no-non-null-assertion
            expect(root).toHaveDisplayName(ButtonGroup.displayName!);

            expect(group).toExist();
            expect(group).toHaveClassName(CLASSNAME);

            expect(button).toExist();
            expect(iconButton).toExist();
            expect(icon).toExist();
            expect(icon.length).toEqual(2);

            expect(dropdown).not.toExist();
        });
    });

    /////////////////////////////

    // 2. Test defaultProps value and important props custom values.
    describe('Props', (): void => {
        it('should use default props', (): void => {
            const { root }: ISetup = setup();

            // tslint:disable-next-line: no-non-null-assertion
            expect(root).toHaveDisplayName(Button.displayName!);
        });

        it("should use 'button' `variant` whatever the given `variant` prop is", (): void => {
            mockConsole();

            const testedProp: string = 'variant';
            const modifiedProps: ISetupProps = {
                [testedProp]: ButtonVariants.button,
            };

            let { button }: ISetup = setup(modifiedProps);
            expect(button).toHaveProp(testedProp, ButtonVariants.button);

            /////////////////////////////

            modifiedProps[testedProp] = ButtonVariants.button;

            ({ button } = setup(modifiedProps));

            expect(button).toHaveProp(testedProp, ButtonVariants.button);

            /////////////////////////////

            modifiedProps[testedProp] = ButtonVariants.icon;

            ({ button } = setup({ ...modifiedProps, isSplitted: true }));

            expect(button).toHaveProp(testedProp, ButtonVariants.button);

            /////////////////////////////

            modifiedProps[testedProp] = ButtonVariants.button;

            ({ button } = setup(modifiedProps));

            expect(button).toHaveProp(testedProp, ButtonVariants.button);
        });

        it(`should forward any <${Button.displayName}> prop (except \`variant\`)`, (): void => {
            const modifiedPropsBuilder: () => ISetupProps = build('props').fields({
                // tslint:disable-next-line: no-any
                color: fake((fakeData: any) => fakeData.commerce.color()),
                emphasis: oneOf(...Object.values(Emphasises)),
                size: oneOf(...Object.values(Sizes)),
                theme: oneOf(...Object.values(Themes)),
            });

            const modifiedProps: ISetupProps = modifiedPropsBuilder();

            let { button }: ISetup = setup(modifiedProps);

            Object.keys(modifiedProps).forEach((prop: string) => {
                expect(button).toHaveProp(prop, modifiedProps[prop]);
            });

            /////////////////////////////

            ({ button } = setup({ ...modifiedProps, isSplitted: true }));

            Object.keys(modifiedProps).forEach((prop: string) => {
                expect(button).toHaveProp(prop, modifiedProps[prop]);
            });
        });

        it('should forward any other props', (): void => {
            const testedProp: string = 'winter';
            const modifiedProps: ISetupProps = {
                [testedProp]: 'is coming',
            };

            let { button, iconButton }: ISetup = setup(modifiedProps);

            expect(button).toHaveProp(testedProp, modifiedProps[testedProp]);

            /////////////////////////////

            ({ button, iconButton } = setup({ ...modifiedProps, isSplitted: true }));

            expect(button).toHaveProp(testedProp, modifiedProps[testedProp]);
            expect(iconButton).toHaveProp(testedProp, modifiedProps[testedProp]);
        });
    });

    /////////////////////////////

    // 3. Test events.
    describe('Events', (): void => {
        const onClick: jest.Mock = jest.fn();

        beforeEach(
            (): void => {
                onClick.mockClear();
            },
        );

        it('should both trigger `onClick` and toggle the dropdown when the button is clicked in non-splitted mode', () => {
            const setupReturn: ISetup = setup({ onClick }, false);
            const { button, wrapper }: ISetup = setupReturn;
            let { dropdown }: ISetup = setupReturn;

            expect(dropdown).not.toExist();

            button.simulate('click');
            dropdown = wrapper.find('Dropdown');

            expect(onClick).toHaveBeenCalled();
            expect(wrapper.find('Dropdown')).toExist();
        });

        it('should only trigger `onClick` when the label button is clicked in splitted mode', () => {
            const setupReturn: ISetup = setup({ isSplitted: true, onClick }, false);
            const { button, wrapper }: ISetup = setupReturn;
            let { dropdown }: ISetup = setupReturn;

            expect(dropdown).not.toExist();

            button.first().simulate('click');
            dropdown = wrapper.find('Dropdown');

            expect(onClick).toHaveBeenCalled();
            expect(wrapper.find('Dropdown')).not.toExist();
        });

        it('should only toggle the dropdown when the dropdown (icon) button is clicked in splitted mode', () => {
            const setupReturn: ISetup = setup({ isSplitted: true, onClick }, false);
            const { iconButton, wrapper }: ISetup = setupReturn;
            let { dropdown }: ISetup = setupReturn;

            expect(dropdown).not.toExist();

            iconButton.simulate('click');
            dropdown = wrapper.find('Dropdown');

            expect(onClick).not.toHaveBeenCalled();
            expect(wrapper.find('Dropdown')).toExist();
        });
    });

    /////////////////////////////

    // 4. Test conditions (i.e. things that display or not in the UI based on props).
    describe('Conditions', (): void => {
        beforeEach(
            (): void => {
                try {
                    // If `console.warn` has been mocked at least one, this exists. So disable TS here.
                    // @ts-ignore
                    global.console.warn.mockRestore();
                } catch (exception) {
                    // Nothing to do here.
                }
            },
        );

        it('should fail when more than 2 children are given', (): void => {
            let children: React.ReactNode = (
                <Fragment>
                    <Icon icon={mdiPlus} />
                    <Icon icon={mdiPlus} />
                    <span>Label</span>
                </Fragment>
            );

            expect(
                (): void => {
                    setup({ children });
                },
            ).toThrowErrorMatchingSnapshot();

            /////////////////////////////

            children = (
                <Fragment>
                    <Icon icon={mdiPlus} />
                    <Icon icon={mdiPlus} />
                    <span>Label</span>
                </Fragment>
            );

            expect(
                (): void => {
                    setup({ children, isSplitted: true });
                },
            ).toThrowErrorMatchingSnapshot();

            /////////////////////////////

            children = (
                <Fragment>
                    <Icon icon={mdiPlus} />
                    <span>Label</span>
                    <span>Label 2</span>
                </Fragment>
            );

            expect(
                (): void => {
                    setup({ children });
                },
            ).toThrowErrorMatchingSnapshot();

            /////////////////////////////

            children = (
                <Fragment>
                    <Icon icon={mdiPlus} />
                    <span>Label</span>
                    <span>Label 2</span>
                </Fragment>
            );

            expect(
                (): void => {
                    setup({ children, isSplitted: true });
                },
            ).toThrowErrorMatchingSnapshot();
        });

        it(`should warn the user when rendering a <${DropdownButton.displayName}> with a \`variant\``, (): void => {
            global.console.warn = jest.fn();

            // We know that a <DropdownButton> cannot receive a `variant`, but for the purpose of the test ignore it.
            // @ts-ignore
            setup({ variant: ButtonVariants.icon });
            // tslint:disable-next-line: no-unbound-method
            expect(global.console.warn).toHaveBeenCalled();

            // @ts-ignore
            global.console.warn.mockClear();

            // We know that a <DropdownButton> cannot receive a `variant`, but for the purpose of the test ignore it.
            // @ts-ignore
            setup({ variant: ButtonVariants.button });
            // tslint:disable-next-line: no-unbound-method
            expect(global.console.warn).toHaveBeenCalled();
        });

        it(`should warn the user when rendering a <${DropdownButton.displayName}> without a \`dropdown\``, (): void => {
            global.console.warn = jest.fn();

            // We know that a <DropdownButton> must receive a `dropdown`, but for the purpose of the test ignore it.
            // @ts-ignore
            // tslint:disable-next-line: no-null-keyword
            setup({ dropdown: null });
            // tslint:disable-next-line: no-unbound-method
            expect(global.console.warn).toHaveBeenCalled();

            // @ts-ignore
            global.console.warn.mockClear();

            // We know that a <DropdownButton> must receive a `dropdown`, but for the purpose of the test ignore it.
            // @ts-ignore
            // tslint:disable-next-line: no-null-keyword
            setup({ dropdown: null });
            // tslint:disable-next-line: no-unbound-method
            expect(global.console.warn).toHaveBeenCalled();
        });
    });

    /////////////////////////////

    // 5. Test state.
    describe('State', (): void => {
        // Nothing to do here.
    });

    /////////////////////////////

    // Common tests suite.
    commonTestsSuite(setup, { className: 'root' }, { className: CLASSNAME });
});
