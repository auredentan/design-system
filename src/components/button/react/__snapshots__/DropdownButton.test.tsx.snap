// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropdownButton> Conditions should fail when more than 2 children are given 1`] = `"<DropdownButton> cannot have more than 2 children (got 3)!"`;

exports[`<DropdownButton> Conditions should fail when more than 2 children are given 2`] = `"<DropdownButton> cannot have more than 2 children (got 3)!"`;

exports[`<DropdownButton> Conditions should fail when more than 2 children are given 3`] = `"<DropdownButton> cannot have more than 2 children (got 3)!"`;

exports[`<DropdownButton> Conditions should fail when more than 2 children are given 4`] = `"<DropdownButton> cannot have more than 2 children (got 3)!"`;

exports[`<DropdownButton> Conditions should fail when using \`leftIcon\` instead of \`icon\` 1`] = `"Pre-validation of <DropdownButton> failed: 'You must use the \`icon\` prop of <DropdownButton> instead of \`leftIcon\`!'."`;

exports[`<DropdownButton> Snapshots and structure should render correctly a non-splitted dropdown button 1`] = `
<DropdownButton
  dropdown={
    <span>
      Dropdown content
    </span>
  }
>
  <Dropdown
    showDropdown={false}
    toggleElement={
      <Button
        className="lumx-dropdown-button"
        onClick={[Function]}
        rightIcon="M7,10L12,15L17,10H7Z"
        variant="button"
      >
        Label
      </Button>
    }
  >
    <div
      className="lumx-dropdown lumx-dropdown--has-toggle"
      onKeyDown={[Function]}
    >
      <Popover
        anchorElement={
          <div
            className="lumx-dropdown__toggle"
            onClick={[Function]}
          >
            <Button
              className="lumx-dropdown-button"
              onClick={[Function]}
              rightIcon="M7,10L12,15L17,10H7Z"
              variant="button"
            >
              Label
            </Button>
          </div>
        }
        popperElement={false}
        showPopper={false}
      >
        <Manager>
          <Reference>
            <InnerReference
              setReferenceNode={[Function]}
            >
              <div
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "height": "fit-content",
                    "width": "fit-content",
                  }
                }
              >
                <div
                  className="lumx-dropdown__toggle"
                  onClick={[Function]}
                >
                  <Button
                    className="lumx-dropdown-button"
                    onClick={[Function]}
                    rightIcon="M7,10L12,15L17,10H7Z"
                    variant="button"
                  >
                    <ButtonRoot
                      className="lumx-dropdown-button lumx-button--has-right-icon lumx-button lumx-button--color-primary lumx-button--emphasis-high lumx-button--size-m lumx-button--theme-light lumx-button--variant-button"
                      onClick={[Function]}
                    >
                      <button
                        className="lumx-dropdown-button lumx-button--has-right-icon lumx-button lumx-button--color-primary lumx-button--emphasis-high lumx-button--size-m lumx-button--theme-light lumx-button--variant-button"
                        onClick={[Function]}
                      >
                        <span>
                          Label
                        </span>
                        <Icon
                          icon="M7,10L12,15L17,10H7Z"
                        >
                          <i
                            className="lumx-icon"
                          >
                            <svg
                              aria-hidden="true"
                              height="1em"
                              preserveAspectRatio="xMidYMid meet"
                              style={
                                Object {
                                  "verticalAlign": "-0.125em",
                                }
                              }
                              viewBox="0 0 24 24"
                              width="1em"
                            >
                              <path
                                d="M7,10L12,15L17,10H7Z"
                                fill="currentColor"
                              />
                            </svg>
                          </i>
                        </Icon>
                      </button>
                    </ButtonRoot>
                  </Button>
                </div>
              </div>
            </InnerReference>
          </Reference>
        </Manager>
      </Popover>
    </div>
  </Dropdown>
</DropdownButton>
`;

exports[`<DropdownButton> Snapshots and structure should render correctly a non-splitted dropdown button with a preceding icon 1`] = `
<DropdownButton
  dropdown={
    <span>
      Dropdown content
    </span>
  }
  icon="M19,13H13V19H11V13H5V11H11V5H13V11H19V13Z"
>
  <Dropdown
    showDropdown={false}
    toggleElement={
      <Button
        className="lumx-dropdown-button"
        leftIcon="M19,13H13V19H11V13H5V11H11V5H13V11H19V13Z"
        onClick={[Function]}
        rightIcon="M7,10L12,15L17,10H7Z"
        variant="button"
      >
        Label
      </Button>
    }
  >
    <div
      className="lumx-dropdown lumx-dropdown--has-toggle"
      onKeyDown={[Function]}
    >
      <Popover
        anchorElement={
          <div
            className="lumx-dropdown__toggle"
            onClick={[Function]}
          >
            <Button
              className="lumx-dropdown-button"
              leftIcon="M19,13H13V19H11V13H5V11H11V5H13V11H19V13Z"
              onClick={[Function]}
              rightIcon="M7,10L12,15L17,10H7Z"
              variant="button"
            >
              Label
            </Button>
          </div>
        }
        popperElement={false}
        showPopper={false}
      >
        <Manager>
          <Reference>
            <InnerReference
              setReferenceNode={[Function]}
            >
              <div
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "height": "fit-content",
                    "width": "fit-content",
                  }
                }
              >
                <div
                  className="lumx-dropdown__toggle"
                  onClick={[Function]}
                >
                  <Button
                    className="lumx-dropdown-button"
                    leftIcon="M19,13H13V19H11V13H5V11H11V5H13V11H19V13Z"
                    onClick={[Function]}
                    rightIcon="M7,10L12,15L17,10H7Z"
                    variant="button"
                  >
                    <ButtonRoot
                      className="lumx-dropdown-button lumx-button--has-left-icon lumx-button--has-right-icon lumx-button lumx-button--color-primary lumx-button--emphasis-high lumx-button--size-m lumx-button--theme-light lumx-button--variant-button"
                      onClick={[Function]}
                    >
                      <button
                        className="lumx-dropdown-button lumx-button--has-left-icon lumx-button--has-right-icon lumx-button lumx-button--color-primary lumx-button--emphasis-high lumx-button--size-m lumx-button--theme-light lumx-button--variant-button"
                        onClick={[Function]}
                      >
                        <Icon
                          icon="M19,13H13V19H11V13H5V11H11V5H13V11H19V13Z"
                        >
                          <i
                            className="lumx-icon"
                          >
                            <svg
                              aria-hidden="true"
                              height="1em"
                              preserveAspectRatio="xMidYMid meet"
                              style={
                                Object {
                                  "verticalAlign": "-0.125em",
                                }
                              }
                              viewBox="0 0 24 24"
                              width="1em"
                            >
                              <path
                                d="M19,13H13V19H11V13H5V11H11V5H13V11H19V13Z"
                                fill="currentColor"
                              />
                            </svg>
                          </i>
                        </Icon>
                        <span>
                          Label
                        </span>
                        <Icon
                          icon="M7,10L12,15L17,10H7Z"
                        >
                          <i
                            className="lumx-icon"
                          >
                            <svg
                              aria-hidden="true"
                              height="1em"
                              preserveAspectRatio="xMidYMid meet"
                              style={
                                Object {
                                  "verticalAlign": "-0.125em",
                                }
                              }
                              viewBox="0 0 24 24"
                              width="1em"
                            >
                              <path
                                d="M7,10L12,15L17,10H7Z"
                                fill="currentColor"
                              />
                            </svg>
                          </i>
                        </Icon>
                      </button>
                    </ButtonRoot>
                  </Button>
                </div>
              </div>
            </InnerReference>
          </Reference>
        </Manager>
      </Popover>
    </div>
  </Dropdown>
</DropdownButton>
`;

exports[`<DropdownButton> Snapshots and structure should render correctly a splitted dropdown button 1`] = `
<DropdownButton
  dropdown={
    <span>
      Dropdown content
    </span>
  }
  splitted={true}
>
  <Dropdown
    showDropdown={false}
    toggleElement={
      <ButtonGroup
        className="lumx-dropdown-button lumx-dropdown-button--is-splitted"
      >
        <Button
          variant="button"
        >
          Label
        </Button>
        <IconButton
          icon="M7,10L12,15L17,10H7Z"
          onClick={[Function]}
        />
      </ButtonGroup>
    }
  >
    <div
      className="lumx-dropdown lumx-dropdown--has-toggle"
      onKeyDown={[Function]}
    >
      <Popover
        anchorElement={
          <div
            className="lumx-dropdown__toggle"
            onClick={[Function]}
          >
            <ButtonGroup
              className="lumx-dropdown-button lumx-dropdown-button--is-splitted"
            >
              <Button
                variant="button"
              >
                Label
              </Button>
              <IconButton
                icon="M7,10L12,15L17,10H7Z"
                onClick={[Function]}
              />
            </ButtonGroup>
          </div>
        }
        popperElement={false}
        showPopper={false}
      >
        <Manager>
          <Reference>
            <InnerReference
              setReferenceNode={[Function]}
            >
              <div
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "height": "fit-content",
                    "width": "fit-content",
                  }
                }
              >
                <div
                  className="lumx-dropdown__toggle"
                  onClick={[Function]}
                >
                  <ButtonGroup
                    className="lumx-dropdown-button lumx-dropdown-button--is-splitted"
                  >
                    <div
                      className="lumx-dropdown-button lumx-dropdown-button--is-splitted lumx-button-group"
                    >
                      <Button
                        variant="button"
                      >
                        <ButtonRoot
                          className="lumx-button lumx-button--color-primary lumx-button--emphasis-high lumx-button--size-m lumx-button--theme-light lumx-button--variant-button"
                        >
                          <button
                            className="lumx-button lumx-button--color-primary lumx-button--emphasis-high lumx-button--size-m lumx-button--theme-light lumx-button--variant-button"
                          >
                            <span>
                              Label
                            </span>
                          </button>
                        </ButtonRoot>
                      </Button>
                      <IconButton
                        icon="M7,10L12,15L17,10H7Z"
                        onClick={[Function]}
                      >
                        <Button
                          className="lumx-icon-button"
                          leftIcon="M7,10L12,15L17,10H7Z"
                          onClick={[Function]}
                          variant="icon"
                        >
                          <ButtonRoot
                            className="lumx-icon-button lumx-button lumx-button--color-primary lumx-button--emphasis-high lumx-button--size-m lumx-button--theme-light lumx-button--variant-icon"
                            onClick={[Function]}
                          >
                            <button
                              className="lumx-icon-button lumx-button lumx-button--color-primary lumx-button--emphasis-high lumx-button--size-m lumx-button--theme-light lumx-button--variant-icon"
                              onClick={[Function]}
                            >
                              <Icon
                                icon="M7,10L12,15L17,10H7Z"
                              >
                                <i
                                  className="lumx-icon"
                                >
                                  <svg
                                    aria-hidden="true"
                                    height="1em"
                                    preserveAspectRatio="xMidYMid meet"
                                    style={
                                      Object {
                                        "verticalAlign": "-0.125em",
                                      }
                                    }
                                    viewBox="0 0 24 24"
                                    width="1em"
                                  >
                                    <path
                                      d="M7,10L12,15L17,10H7Z"
                                      fill="currentColor"
                                    />
                                  </svg>
                                </i>
                              </Icon>
                            </button>
                          </ButtonRoot>
                        </Button>
                      </IconButton>
                    </div>
                  </ButtonGroup>
                </div>
              </div>
            </InnerReference>
          </Reference>
        </Manager>
      </Popover>
    </div>
  </Dropdown>
</DropdownButton>
`;

exports[`<DropdownButton> Snapshots and structure should render correctly a splitted dropdown button with a preceding icon 1`] = `
<DropdownButton
  dropdown={
    <span>
      Dropdown content
    </span>
  }
  icon="M19,13H13V19H11V13H5V11H11V5H13V11H19V13Z"
  splitted={true}
>
  <Dropdown
    showDropdown={false}
    toggleElement={
      <ButtonGroup
        className="lumx-dropdown-button lumx-dropdown-button--is-splitted"
      >
        <Button
          leftIcon="M19,13H13V19H11V13H5V11H11V5H13V11H19V13Z"
          variant="button"
        >
          Label
        </Button>
        <IconButton
          icon="M7,10L12,15L17,10H7Z"
          onClick={[Function]}
        />
      </ButtonGroup>
    }
  >
    <div
      className="lumx-dropdown lumx-dropdown--has-toggle"
      onKeyDown={[Function]}
    >
      <Popover
        anchorElement={
          <div
            className="lumx-dropdown__toggle"
            onClick={[Function]}
          >
            <ButtonGroup
              className="lumx-dropdown-button lumx-dropdown-button--is-splitted"
            >
              <Button
                leftIcon="M19,13H13V19H11V13H5V11H11V5H13V11H19V13Z"
                variant="button"
              >
                Label
              </Button>
              <IconButton
                icon="M7,10L12,15L17,10H7Z"
                onClick={[Function]}
              />
            </ButtonGroup>
          </div>
        }
        popperElement={false}
        showPopper={false}
      >
        <Manager>
          <Reference>
            <InnerReference
              setReferenceNode={[Function]}
            >
              <div
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "height": "fit-content",
                    "width": "fit-content",
                  }
                }
              >
                <div
                  className="lumx-dropdown__toggle"
                  onClick={[Function]}
                >
                  <ButtonGroup
                    className="lumx-dropdown-button lumx-dropdown-button--is-splitted"
                  >
                    <div
                      className="lumx-dropdown-button lumx-dropdown-button--is-splitted lumx-button-group"
                    >
                      <Button
                        leftIcon="M19,13H13V19H11V13H5V11H11V5H13V11H19V13Z"
                        variant="button"
                      >
                        <ButtonRoot
                          className="lumx-button--has-left-icon lumx-button lumx-button--color-primary lumx-button--emphasis-high lumx-button--size-m lumx-button--theme-light lumx-button--variant-button"
                        >
                          <button
                            className="lumx-button--has-left-icon lumx-button lumx-button--color-primary lumx-button--emphasis-high lumx-button--size-m lumx-button--theme-light lumx-button--variant-button"
                          >
                            <Icon
                              icon="M19,13H13V19H11V13H5V11H11V5H13V11H19V13Z"
                            >
                              <i
                                className="lumx-icon"
                              >
                                <svg
                                  aria-hidden="true"
                                  height="1em"
                                  preserveAspectRatio="xMidYMid meet"
                                  style={
                                    Object {
                                      "verticalAlign": "-0.125em",
                                    }
                                  }
                                  viewBox="0 0 24 24"
                                  width="1em"
                                >
                                  <path
                                    d="M19,13H13V19H11V13H5V11H11V5H13V11H19V13Z"
                                    fill="currentColor"
                                  />
                                </svg>
                              </i>
                            </Icon>
                            <span>
                              Label
                            </span>
                          </button>
                        </ButtonRoot>
                      </Button>
                      <IconButton
                        icon="M7,10L12,15L17,10H7Z"
                        onClick={[Function]}
                      >
                        <Button
                          className="lumx-icon-button"
                          leftIcon="M7,10L12,15L17,10H7Z"
                          onClick={[Function]}
                          variant="icon"
                        >
                          <ButtonRoot
                            className="lumx-icon-button lumx-button lumx-button--color-primary lumx-button--emphasis-high lumx-button--size-m lumx-button--theme-light lumx-button--variant-icon"
                            onClick={[Function]}
                          >
                            <button
                              className="lumx-icon-button lumx-button lumx-button--color-primary lumx-button--emphasis-high lumx-button--size-m lumx-button--theme-light lumx-button--variant-icon"
                              onClick={[Function]}
                            >
                              <Icon
                                icon="M7,10L12,15L17,10H7Z"
                              >
                                <i
                                  className="lumx-icon"
                                >
                                  <svg
                                    aria-hidden="true"
                                    height="1em"
                                    preserveAspectRatio="xMidYMid meet"
                                    style={
                                      Object {
                                        "verticalAlign": "-0.125em",
                                      }
                                    }
                                    viewBox="0 0 24 24"
                                    width="1em"
                                  >
                                    <path
                                      d="M7,10L12,15L17,10H7Z"
                                      fill="currentColor"
                                    />
                                  </svg>
                                </i>
                              </Icon>
                            </button>
                          </ButtonRoot>
                        </Button>
                      </IconButton>
                    </div>
                  </ButtonGroup>
                </div>
              </div>
            </InnerReference>
          </Reference>
        </Manager>
      </Popover>
    </div>
  </Dropdown>
</DropdownButton>
`;
